source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,597 @@
+
+Furthermore, security groups allow these rules to be applied in fine
+grained fashion - e.g. the user can specify which rules apply to which
+ports.
+
+Below, you can find some use cases of this feature:
+* **only** allow incoming traffic to a specific VM from a specific
+CIDR
+* **only** allow incoming access to a range of destination ports in a
+set of VMs
+* **only** allow outgoing traffic to a specific port on a set of VMs
+
+L2 protection is **not in scope** of the Networking API security groups.
+
+## Objectives
+
+* Provide a security group implementation, using OVN as the backend.
+No additional backend / database should be used other than ovn's
+northbound database.
+
+* When possible, Openstack neutron's behavior should be mimicked: by
+default, all VM incoming IP *and* outgoing traffic is dropped. This behavior
+is better specified in the [requirements](#requirements) section.
+
+### Requirements
+
+* For ingress traffic (to an instance)
+  + Only traffic matched with security group rules is allowed.
+  + When there are no rules defined, all ingress traffic is dropped.
+* For egress traffic (from an instance)
+  + Only traffic matched with security group rules is allowed.
+  + When there are no rules defined, all egress traffic is dropped.
+  + Rules allowing all egress IP traffic are automatically added to all groups.
+* The DHCP traffic for OVN subnets will have to be explicitly allowed,
+otherwise, it will be automatically dropped, as a consequence to the above requirements.
+
+### Security groups description
+
+Security groups, in the general sense, are meant to group together a list of
+[security group rules](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules). 
+They are applied to logical ports, and thus are used to associate a set of 
+rules to a port.
+
+They are are indeed a white list - e.g. everything is *denied* by
+default, and the user adds rules that allow traffic. As long as *one*
+rule is matched, the traffic is allowed.
+
+Security group rules can match on L3 and L4 parameters, specifically on
+the L3 protocol, the L4 protocol, and destination ports. The remote IP
+address is also a subject for the match conditions, and rules can be
+added to allow *incoming* and *outgoing* traffic.
+
+The API of security group rules can be found [here](https://developer.openstack.org/api-ref/network/v2/#security-group-rules-security-group-rules).
+
+Security group filters are **only applied to** ports whose
+**port_security_enabled** attribute is set. By default, when a new port is
+created, and the user does not specify this attribute, it is created set
+to 'True', meaning that security group filters will be applied to it.
+
+This causes all incomming traffic to the port to be dropped, until the user
+white-lists the intended traffic for that port.
+
+### User interface
+
+The user is meant to provision security group(s) and rule(s) through ansible,
```

Comment:
"through ansible" - how? contacting the host? provider?

Reply:
Good point; assumptions are really bad.

Done